impress-org/give-google-analytics
# Feature: Add Google Analytic 4 tracking on v3 forms

## Description
This PR introduces a Google analytics as a field for all v3 forms. It will load an empty element at the bottom of each form and render the following modified gtags for GA4 tracking + all the predefined recommended tags from GA4. This GA field will auto-populate on any new v3 form without user engagement if they have GA4 settings set up by hooking into the givewp_donation_form_schema action.

```
  gtag("event", "page_view", {
        page_title: window.parent.document.title,
    });

   gtag('event', 'view_item', {
                    currency: currency,
                    value: donationAmount,
                    items: [
                        {
                            item_id: formId,
                            item_name: name,
                            item_brand: 'Fundraising',
                            affiliation: affiliation ,
                            item_category:  trackCategory,
                            item_list_name: trackListName ,
                        }
                    ]
                })
```
```
    gtag('event', 'begin_checkout', {
                        currency: currency,
                        value: donationAmount,
                        items: [
                            {
                                item_id: formId,
                                item_name: name,
                                item_brand: 'Fundraising',
                                affiliation: affiliation ,
                               item_category:  trackCategory,
                                item_category2: gateway,
                                item_category3: donationType,
                                item_list_name: trackListName ,
                                price: donationAmount,
                                quantity: 1
                            }
                        ]
                    });
```

## Visuals
<img width="1723" alt="Screen Shot 2023-08-08 at 2 40 03 PM" src="https://github.com/impress-org/give-google-analytics/assets/75056371/51c90091-ff0f-4d7f-a8ec-97aa92cf5cbc">

## Testing Instructions
- Create a GA4 account through google. You will need to create and connect a real webpage with a .com in order to start receiving data (I used the QA page). 
- Once connected activate the plugin and go to the global settings Donations > Settings > Google analytics
& Enter your config details
- Create a v3 form and view/submit a donation on the form.
- Verify event and event data is sent and received with RealTime reports on GoogleAnalytics
- I found this chrome extension useful to watch the gtag commands : https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna


## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205251844689198
